R/plotstyle.R

```diff
@@ -1,174 +1,194 @@
-#' @param out Switch defining which type of plot style you want to get: Currently you can choose between "color", "legend" and "all" (the latter will return a dataframe with all available plot styles)
-#' @param unknown Optional data frame defining plot styles for unknown entities. A default color map will be used for unknown entities if nothing is provided here
-#' @param plot If TRUE plots with all given entities and their colors will be produced (to produce plots with all available entitities leave the \code{...} entry empty!)
-#' @param verbosity Set to 1 if you want to know for which unknown entities plotstyle brewed colors
-#' @param regexp If \code{TRUE}, match entities as regular expressions. 
-#'        Matching entities are expanded, non-matching entities are returned as 
-#'        the original expression.  Does not generate default colour maps. 
-#'        Implies \code{plot = FALSE} and \code{verbosity = 0}. 
+#'
+#' Returns a named vector (using entity names) with style codes (e.g. colors)
+#' for given entities.
+#'
+#' @param ... One or more strings or a vector of strings with names of entities
+#'   (regions, variable names, etc.). Units in brackets "(US$2005/GJ)" will be
+#'   ignored. If left empty all available entities will be used
+#' @param out Switch defining which type of plot style you want to get:
+#'   Currently you can choose between "color", "legend" and "all" (the latter
+#'   will return a dataframe with all available plot styles)
+#' @param unknown Optional data frame defining plot styles for unknown entities.
+#'   A default color map will be used for unknown entities if nothing is
+#'   provided here
+#' @param plot If TRUE plots with all given entities and their colors will be
+#'   produced (to produce plots with all available entities leave the \code{...}
+#'   entry empty!)
+#' @param verbosity Set to 1 if you want to know for which unknown entities
+#'   plotstyle brewed colors
+#' @param regexp If \code{TRUE}, match entities as regular expressions. Matching
+#'   entities are expanded, non-matching entities are returned as the original
+#'   expression.  Does not generate default color maps. Implies \code{plot =
+#'   FALSE} and \code{verbosity = 0}.
 #' @return Plot styles for given entities
+#' @section Colors for unknown entities:
+#' \if{html}{\figure{colors.png}{options: width="100\%"}}
 #' @author David Klein, Jan Philipp Dietrich
 #' @seealso \code{\link{plotstyle.add}}
 #' @examples
-#' entities <- c("AFR","AAA","AFR","UNKNOWN_ELEMENT2")
+#' entities <- c("AFR", "AAA", "AFR", "UNKNOWN_ELEMENT2")
 #' plotstyle(entities)
-#' unknown=data.frame(row.names=c("AAA","UNKNOWN_ELEMENT2"),
-#'                    color=c("#123456","#345678"),
-#'                    legend=c("l_AAA","l_UNKNOWN_ELEMENT2"))
-#' plotstyle(entities,unknown=unknown)
-#' plotstyle(entities,out="legend")
-#' plotstyle(entities,out="all")
-#' 
+#' unknown <- data.frame(row.names = c("AAA", "UNKNOWN_ELEMENT2"),
```

Comment:
thanks for all this 'beauty' work – but it sometimes makes it a bit difficult to see where you actually changed something in the code ;)

Reply:
I tried to follow the coding guideline to keep to the style guide and potentially fix styling issues as indicated by `lucode2::lint()`, e.g., by using `lucode2::autoFormat()`. But I see your point... 
Ideally, style is only improved and new code is committed with good code style. Then there will only be a finite number of such confusing style fixing commits ;-)

Reply:
Good point! I added links in the initial description.